lib/my_inject.rb

```diff
@@ -0,0 +1,41 @@
+class Array
+	
+	def my_inject(accumulator = nil, operation = nil, &block)
+		if accumulator.nil? && operation.nil? && block.nil?
+			raise ArgumentError, "Please provide either an operation or a block"
+		end
+
+		if operation && block
+			raise ArgumentError, "Please provide either an operation symbol or a block, not both"
```

Comment:
Line is too long. [88/80]<br>Use 2 (not 1) spaces for indentation.<br>Tab detected.